compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/optimizer/LargeMethodOptimizer.java

```diff
@@ -134,16 +144,17 @@ private boolean hasLessInstructionCount(BIRFunction birFunction) {
             BIRTerminator bbTerminator = basicBlock.terminator;
             if (splitStarted) {
                 if (bbTerminator.lhsOp != null) {
-                    // as the varDcl is available inside the split no need to pass as function arg
-                    if (needToPassLhsVarDclAsArg(bbTerminator.lhsOp)) {
+                    if ((bbTerminator.lhsOp.variableDcl.kind == VarKind.LOCAL) ||
+                            (bbTerminator.lhsOp.variableDcl.kind == VarKind.RETURN)) {
+                        // if a local or the return var is assigned value in a BB terminator, no split is done
+                        splitStarted = false;
```

Comment:
Why did we consider VarKind.LOCAL here?

Reply:
We cannot assign value for a local variable inside a new split function, as they should be passed as arguments. So if `VarKind.LOCAL` var is assigned value, no split is done. Here I think `bbTerminator.lhsOp.variableDcl.kind == VarKind.RETURN` won't happen most probably. But added that condition for the safety.